packages/client/src/fs/endpoints/defaultEndpoint.ts

```diff
@@ -0,0 +1,60 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2024 TypeFox and others.
+ * Licensed under the MIT License. See LICENSE in the package root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Logger } from 'monaco-languageclient/tools';
+import { DirectoryListingRequest, DirectoryListingRequestResult, EndpointType, FileReadRequest, FileReadRequestResult, FileSystemEndpoint, FileUpdate, FileUpdateResult, StatsRequest, StatsRequestResult } from '../definitions.js';
+
+export class EmptyFileSystemEndpoint implements FileSystemEndpoint {
+
+    private endpointType: EndpointType;
+    private logger?: Logger;
+
+    constructor(endpointType: EndpointType) {
+        this.endpointType = endpointType;
+    }
+
+    init(): void { }
+
+    getFileSystemInfo(): string {
+        return 'This file system performs no operations.';
+    }
+
+    setLogger(logger: Logger): void {
+        this.logger = logger;
+    }
+
+    getEndpointType(): EndpointType {
+        return this.endpointType;
+    }
+
+    readFile(params: FileReadRequest): Promise<FileReadRequestResult> {
+        this.logger?.info(`Reading file: ${params.resourceUri}`);
+        return Promise.resolve({
+            status: 'denied',
```

Comment:
any reason why it's not throwing an error?

Reply:
Because the status carries the info. It could be a `Promise.reject`, too. The whole file system / file handling realm will still evolve, so if it is not critical I will leave it as is.